github4s/src/test/scala/github4s/integration/GHActivitiesSpec.scala

```diff
@@ -16,19 +16,20 @@
 
 package github4s.integration
 
+import cats.effect.IO
+import github4s.GithubIOSyntax._
 import github4s.Github
-import github4s.Github._
-import github4s.free.domain.{Stargazer, StarredRepository, Subscription}
-import github4s.implicits1._
+import github4s.domain._
+
 import github4s.utils.{BaseIntegrationSpec, Integration}
 
 trait GHActivitiesSpec extends BaseIntegrationSpec {
 
   "Activity >> Set a thread subscription" should "return expected response when a valid thread id is provided" taggedAs Integration in {
     val response =
-      Github(accessToken).activities
-        .setThreadSub(validThreadId, true, false)
-        .execFuture(headerUserAgent)
+      Github[IO](accessToken).activities
+        .setThreadSub(validThreadId, true, false, headerUserAgent)
+        .toFuture
```

Comment:
Mhm, could we use `.attempt` instead?

Then the function: `testFutureIsLeft ` could perform the unsafeRunSync and check the result.

Note: this is valid just in case we want to avoid `Future` usage, as it's discouraged

Reply:
`GHResponse[A]` is a type alias for `Either[GHException, GHResult[A]]`

Using `.attempt` would nest the `Either`s
ex. 
response: IO[Either[Throwable, Either[GHException, GHResult[A]]]]

I could just run `.unsafeRunSync` on the `response` to check if it is `Left` or `Right` and then check the assertions.

Would it be desirable to run all these HTTP Requests synchronously instead of asynchronously?

Reply:
As I mentioned in [this comment](https://github.com/47deg/github4s/pull/344#discussion_r381007470) I'm also inclined to rewrite the examples to use `IO` exclusively. 
And only a side note to mention that we don't recommend to use `Future` as the concurrent effect, but in the case the user wants to keep it, we are providing the `LiftIO[Future]`.

Reply:
The examples are using IO. Question is, should the integration tests also use IO instead of Future.

Reply:
IMHO at some point it might be desired. However, it also could be tackled on a different PR ;)